File: dotnet/cafdemalihapa/Directories/Hosting/Feature/BackEnd/BackEndSecondary/BackEndSecondaryDirectory.cs
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Hosting.Feature.BackEnd.BackEndSecondary
{
    public class BackEndSecondaryDirectory(
            SecondaryApplication secondaryApplication
        )
    {
        public string GetName()
        {
            var backEndDirectoryName = BackEndDirectory.GetName();
            var secondaryApplicationName = secondaryApplication.GetName();
            var name = $"{backEndDirectoryName}.{secondaryApplicationName}";
            return name;
        }

        public string GetPath()
        {
            var backEndDirectoryPath = BackEndDirectory.GetPath();
            var name = GetName();

            var backEndSecondaryDirectoryPath = Path.Combine(backEndDirectoryPath, name);
            return backEndSecondaryDirectoryPath;
        }

        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/BackEnd/BackEndSecondary/BackEndSecondaryDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.BackEnd.BackEndSecondary

open System.IO
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Hosting.Feature.BackEnd

type BackEndSecondaryDirectory(secondaryApplication: SecondaryApplication) =
    member _.GetName() =
        let backEndDirectoryName = BackEndDirectory.GetName()
        let secondaryApplicationName = secondaryApplication.GetName()
        let name = $"{backEndDirectoryName}.{secondaryApplicationName}"
        name

    member this.GetPath() =
        let backEndDirectoryPath = BackEndDirectory.GetPath()
        let name = this.GetName()
        let backEndSecondaryDirectoryPath = Path.Combine(backEndDirectoryPath, name)
        backEndSecondaryDirectoryPath

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore
